### DISCUSSION TITLE
类中某个属性改变时，整个类也得到通知，该如何优雅的处理这个问题

### DISCUSSION THREAD

**Discussion by Vince0911** [BYJRK/DotNet-Discussions#230] [MVVM]
举个例子，我有一个类People，其中属性有Name，Age，Height等等属性，此外在ViewModel中声明了一个PeopleProp属性（类型为People），前台和PeopleProp进行了绑定。
现在我希望People中任一属性发生改变时，例如PeopleProp.Name = "B"，ViewModel中的PeopleProp也能通知前台，目前是通知不了的，因为PeopleProp.Name = "B"没有触发PeopleProp的set方法
目前实现的方式是在Name的set方法中发送WeakReferenceMessenger.Default.Send(new PropertyChangedMessage<string>(this, nameof(Name), default, value), ItemGuid); 
在People的构造函数中注册PropertyChangedMessage<string,Guid>消息（guid用作token）,这样每次Name发生改变时会发送PropertyChangedMessage<string>的消息，被People捕捉到，然后在消息处理方法中，WeakReferenceMessenger.Default.Send(this, ItemGuid);  目的发送People给ViewModel ，
ViewModel里注册了这个消息          
            WeakReferenceMessenger.Default.Register<People, Guid>(this, ItemGuid, (_, m) =>
            {
                PeopleProp = m; // 这里实现PeopleProp的通知
            });
      
 很显然，这种处理方法相当于一种消息的传递接力，太过于复杂和庞大， 看了一下toolkit文档里面也没有收获，OnPropertyChanged()只适用于同一个类中的属性间（同级）通知，有没有更优雅的方式处理这类场景？

**Comment by BYJRK** [comment]
没有好办法。除了用 `Messenger`，你还可以在 VM 里面注册 `PeopleProp.PropertyChanged` 事件。

`BindingList` 可以做到集合中任意元素触发 `PropertyChanged` 事件时自动触发集合的 `OnListChanged` 事件，其实内部也是这个原理。在给集合添加元素的时候，会自动判断这个元素是否实现了 `INotifyPropertyChanged` 接口；如果是，则注册这个事件。

**Comment by qsgy** [comment]
我记得有插件可以自动给你添加属性通知.
但我觉得仍然不够优雅
现在我希望根治问题,打算造一个C#的附庸语言
